staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -683,6 +706,17 @@ func (qs *queueSet) selectQueueLocked() *queue {
 			}
 		}
 	}
+
+	// TODO: add a method to fifo that lets us peek at the oldest request
+	var oldestReqFromMinQueue *request
+	minQueue.requests.Walk(func(r *request) bool {
+		oldestReqFromMinQueue = r
+		return false
+	})
+	if !qs.canAccomodateSeatsLocked(oldestReqFromMinQueue.Seats()) {
+		return nil
+	}
+
```

Comment:
I wrote a long comment why it doesn't work and then I realized it works because we don't modify robinIndex below (we don't modify if we don't dispatch anything).
Can you maybe add an explicit comment above that it's expected and we do that to preserve round-robin behavior?

Reply:
yes, I added a comment, and also the test I added verifies expected value of robin index with multiple calls to `selectQueueLocked`

Reply:
Thanks!